### ISSUE TITLE
Multiple use of ISO_FORTRAN_ENV causes flang to error

### ISSUE THREAD

**Issue by RichBarton-Arm** [flang-compiler/flang#896]
The following example causes flang to emit an error. gfortran handles this fine.

> F90-S-0038-Symbol, compiler_version, has not been explicitly declared (single_source.F90)

This example is reduced from code in the [CP2k application](https://www.cp2k.org/).

As pointed out in the comments, it is the double use of _ISO_FORTRAN_ENV_ via an additional used module with use of _IMPLICIT_NONE_ that causes the issue.

```
MODULE input_section_types
   ! Removing this use statement fixes the issue and flang does the right thing.
   USE ISO_FORTRAN_ENV
END MODULE input_section_types

PROGRAM cp2k
   ! Removing the below use statement makes flang do the right thing.
   use input_section_types
   ! Removing the below line gives the same error (expected)
   USE ISO_FORTRAN_ENV, ONLY: compiler_version
   ! Removing implicit none lets flang build a working example.
   IMPLICIT NONE
   WRITE (*,*) "other: ", compiler_version()
END PROGRAM
```

**Comment by michalpasztamobica** [comment]
I am first getting and information code 0035:
```
(lldb) bt
* thread #1, name = 'flang1', stop reason = breakpoint 4.1
  * frame #0: 0x00000000007d5db1 flang1`error(ecode=I_0035_Predefined_intrinsic_OP1_loses_intrinsic_property, sev=ERR_Informational, eline=13, op1="compiler_version", op2=0x0000000000000000) at error.c:226
    frame #1: 0x000000000072efee flang1`newsym(sptr=607) at semsym.c:1212
    frame #2: 0x00000000006e58cd flang1`ref_intrin(stktop=0x0000000000b43040, list=0x0000000000000001) at semfunc.c:5435
    frame #3: 0x00000000007455cf flang1`mkvarref(stktop=0x0000000000b43040, list=0x0000000000000001) at semutil.c:2127
    frame #4: 0x00000000006894d0 flang1`semant2(rednum=700, top=0x0000000000b43040) at semant2.c:1247
    frame #5: 0x00000000005e3823 flang1`_parser at parser.c:302
    frame #6: 0x00000000005e3266 flang1`parser at parser.c:152
    frame #7: 0x00000000005af7fc flang1`main(argc=2, argv=0x00007fffffffdfa8) at main.c:208
    frame #8: 0x00007ffff7448b97 libc.so.6`__libc_start_main(main=(flang1`main at main.c:147), argc=2, argv=0x00007fffffffdfa8, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffdf98) at libc-start.c:310
    frame #9: 0x000000000041aa0a flang1`_start + 42
```
And later the error itself shows up with the following backtrace:
```
(lldb) bt
* thread #1, name = 'flang1', stop reason = breakpoint 4.1
  * frame #0: 0x00000000007d5db1 flang1`error(ecode=S_0038_Symbol_OP1_has_not_been_explicitly_declared, sev=ERR_Severe, eline=0, op1="compiler_version", op2=0x0000000000000000) at error.c:226
    frame #1: 0x00000000006c98ab flang1`CheckDecl(sptr=674) at semfin.c:3086
    frame #2: 0x00000000006c5ea2 flang1`misc_checks at semfin.c:3448
    frame #3: 0x00000000006be753 flang1`semfin at semfin.c:623
    frame #4: 0x000000000064c8be flang1`semant1(rednum=13, top=0x0000000000b42eb8) at semant.c:1668
    frame #5: 0x00000000005e37db flang1`_parser at parser.c:300
    frame #6: 0x00000000005e3266 flang1`parser at parser.c:152
    frame #7: 0x00000000005af7fc flang1`main(argc=2, argv=0x00007fffffffdfa8) at main.c:208
    frame #8: 0x00007ffff7448b97 libc.so.6`__libc_start_main(main=(flang1`main at main.c:147), argc=2, argv=0x00007fffffffdfa8, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffdf98) at libc-start.c:310
    frame #9: 0x000000000041aa0a flang1`_start + 42
```

**Comment by pawosm-arm** [comment]
Fixed by #1114
